### PR TITLE
feat: Implement the calculation of tokens amounts

### PR DESCRIPTION
**Pull request by nbetsaif**
Implemented dynamic token amount updates based on the toggle switch to calculate the token amount per period or for one cycle, ensuring proper handling of cases where no tokens are available. When tokens are absent, the toggle switch and token-related UI elements are hidden to enhance the user experience by only displaying relevant information.

## Screenshots
PerPeriod            |  For1Cycle
:-------------------------:|:-------------------------:
PerPeriod  |  For1Cycle

### CODE REVIEW

File: lib/ui/proposals/details/components/proposal_details_view.dart [hunk 1]
```diff
@@ -62,312 +56,365 @@ class _ProposalDetailsViewState extends State<ProposalDetailsView> {
...
+    );
+  }
+}
+
+Widget _buildTokenRow(
+    BuildContext context,
+    ProposalDetailsModel proposalDetailsModel,
+    int tokenIndex,
+    String tokenType,
+    bool? isShown) {
```
Comment (n13) [inline lib/ui/proposals/details/components/proposal_details_view.dart:368]: isShown is kind of the wrong name for this parameter

All of this is really funky, I would refactor it with these objectives

1 - Make it clear what is happening and why

2 - Make the calculations explicit and clear, numbers should be calculated with numbers and only converted to strings at display time, by number formatters. 

3 - If there's different types involved, use enums, not plain strings based on UX text.

File: lib/ui/proposals/details/components/proposal_details_view.dart [hunk 2]
```diff
@@ -62,312 +56,365 @@ class _ProposalDetailsViewState extends State<ProposalDetailsView> {
...
+  String? amount;
+
+  // Determine the amount and amountPerPeriod based on token type
+  switch (tokenType) {
+    case 'Utility Token':
+      amount = proposalDetailsModel.tokenValue(0, isShown!);
```
Comment (n13) [inline lib/ui/proposals/details/components/proposal_details_view.dart:374]: tokenValue method should use an enum parameter, also _buildTokenRown should use the same enum

File: lib/core/extension/proposal_details_model_extension.dart [hunk 3]
```diff
@@ -39,27 +43,42 @@ extension ProposalDetailsModelExtension on ProposalDetailsModel {
     return null;
   }
 
-  // TODO(Saif): adjust this function
   String? tokenValue(int index, bool isOneCycleRewardsShown) {
-    String input;
```
Comment (n13) [inline lib/core/extension/proposal_details_model_extension.dart:44]: the model should store all values as numbers

if the values are strings, that's ok, just use accessors to access the number values

there shouldn't be any regexps anywhere here... ??!

**Review by n13 — CHANGES_REQUESTED** [review]
I suggest to refactor the calculation completely

Also cycle is the "larger" value 

1 cycle == 4 periods, as in the web interface

1 cycle = 4 weeks for example

1 cycle = 30 days = 1 moon cycle, 1 period = 1/4 cycle

**Comment by n13** [comment]
Please use these accessors to conver the asset values into doubles - I added computed values to ProposalModelDetails

https://github.com/hypha-dao/hypha-wallet/pull/332

**Review by n13 — APPROVED** [review]
Looking good now

I wonder if cycle duration also comes with settings? 

I saw you're getting period seconds from settings now... maybe cycle seconds are there too? Would be more accurate. 

Once we start testing we need to compare the amounts with the web to make sure it's the same.

**Comment by nbetsaif** [comment]
@n13 cycle duration doesn't exist in the settings. I followed the web implementation, and they set its value statically as i did on the mobile

**Pull request merged**